crates/sui-storage/src/indexes.rs

```diff
@@ -90,6 +104,10 @@ pub struct IndexStoreTables {
     #[default_options_override_fn = "owner_index_table_default_config"]
     owner_index: DBMap<OwnerIndexKey, ObjectInfo>,
 
+    // FIXME use better options
+    #[default_options_override_fn = "owner_index_table_default_config"]
+    coin_index: DBMap<CoinIndexKey, CoinInfo>,
```

Comment:
Would it make any sense to just extend the `owner_index` table with balance? It seems like `CoinInfo` is similar to `ObjectInfo` with missing balance. We would still be able to iterate and filter by coin type and would save on the disk space and still pretty fast(iterate is the key for performance). And perhaps make `ObjectInfo` an enum too to support future evolution

Reply:
owner_index is keyed ` (SuiAddress, ObjectID)` so we can't leverage ordering things by`coin_type`

Reply:
Mu suggestion is we skip `coin_type` in the key. The implication of that is we iterate over all coin types but we can filter the rest from the result when trying to compute balance for a particular type.

Reply:
So we are trading space complexity with time complexity. I think the bottleneck now is time complexity so storing additional data is not as concerning. Anything else makes you think expanding owner_index is better? (the testnet data shows this table is roughly 20GB, let's see if how it grows)